inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/trigger/TriggerManager.java

```diff
@@ -156,6 +169,35 @@ private Runnable jobFetchThread() {
         };
     }
 
+    private boolean isRunningJob(JobProfile profile, Map<String, JobWrapper> jobWrapperMap) {
+        JobWrapper jobWrapper;
+        try {
+            jobWrapper = jobWrapperMap.get(profile.getInstanceId());
+        } catch (Exception exception) {
+            LOGGER.warn("jobWrapper is null");
+            return false;
+        }
+        List<Task> tasks = jobWrapper.getAllTasks();
+        for (Task task : tasks) {
+            JobProfile runJobProfile = task.getJobConf();
+            if (runJobProfile.get(JOB_DIR_FILTER_PATTERN).equals(profile.get(JOB_DIR_FILTER_PATTERN))) {
```

Comment:
Will `jobProfile.get(JOB_DIR_FILTER_PATTERN)` be null?
Wouldn't it be better to use `Objects.equals()` here?

Reply:
done